E2E/tests/voting.test.ts

```diff
@@ -0,0 +1,189 @@
+Feature('voting');
+
+Before(async ({ I, login, username }) => {
+  I.amOnPage('/');
+  login.login(username.user1);
+});
+
+Scenario('Voter can change vote while game is not completed', async ({ I, login, username, roles, game, chessPiece, chessTile }) => {
+  roles.voterLogin();
+  I.waitForText('Waiting for more players');
+  let url = await I.grabCurrentUrl();
+  session(username.user2, () => {
+    login.loginIntoCreatedGameRoom(url, username.user2);
+    roles.voterLogin();
+    I.waitForText('Game in progress');
+    game.vote(chessPiece.pawn, chessTile.a1);
+    game.voteIsVisible(chessPiece.pawn, chessTile.a1, chessPiece.pawnValue);
+    I.waitForElement(game.locator.playersList.playerDoneIcon(username.user2));
+    game.vote(chessPiece.knight, chessTile.e2);
+    game.voteIsNotVisible(chessPiece.pawn, chessTile.a1, chessPiece.pawnValue);
+    game.voteIsVisible(chessPiece.knight, chessTile.e2, chessPiece.knightValue);
+  });
+});
+
+Scenario('Players can’t see each others votes until all of them make a move', async ({ I, login, username, roles, game, chessPiece, chessTile }) => {
+  roles.voterLogin();
+  I.waitForText('Waiting for more players');
+  let url = await I.grabCurrentUrl();
+  session(username.user2, () => {
+    login.loginIntoCreatedGameRoom(url, username.user2);
+    roles.voterLogin();
+    I.waitForText('Game in progress');
+    game.vote(chessPiece.knight, chessTile.e4);
+    game.voteIsVisible(chessPiece.knight, chessTile.e4, chessPiece.knightValue);
+  });
+  I.waitForElement(game.locator.playersList.playerDoneIcon(username.user2));
+  game.voteIsNotVisible(chessPiece.knight, chessTile.e4, chessPiece.knightValue);
+});
+
+Scenario('Two users with same name vote as separate players', async ({ I, login, username, roles, game, chessPiece, chessTile }) => {
+  roles.voterLogin();
+  I.waitForText('Waiting for more players');
+  let url = await I.grabCurrentUrl();
+  session(username.user2, () => {
+    login.loginIntoCreatedGameRoom(url, username.user1);
+    roles.voterLogin();
+    I.waitForText('Game in progress');
```

Comment:
These steps seems to be repeated quite a lot across different tests, maybe we could extract it into a separate function?